### PR TITLE
feat: adding pkg:gcp

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: gcp/example/example.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            fieldTransforms: [
+              FieldTransform(
+                fieldPath: 'count',
+                increment: Value(integerValue: '1'),
+              )
+            ],
+          ),
+        ),
+      ],
+    );
```
Comment (craiglabenz) [inline gcp/example/example.dart:87]: A write to Firestore!

File: gcp/example/example.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    return _Server._(projectId: projectId, client: authClient);
+  }
+
+  final String projectId;
+  final AutoRefreshingAuthClient client;
+
+  late final FirestoreApi api = FirestoreApi(client);
+  late final handler = _onlyGetRootMiddleware.addHandler(_incrementHandler);
```
Comment (craiglabenz) [inline gcp/example/example.dart:43]: I'll have to play around with these `Handler` objects a bit before this all really clicks, I expect

**Review by craiglabenz — COMMENTED** [review]
Already merged, but I still wanted to give it a read

File: gcp/example/example.dart (hunk 1 above)
Comment (kevmoo) [reply gcp/example/example.dart:87]: A transaction, even! This does an atomic increment!

File: gcp/lib/src/terminate.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import 'dart:io';
+
+/// Returns a [Future] that completes when the process receives a
+/// [ProcessSignal] requesting a shutdown.
+///
+/// [ProcessSignal.sigint] is listened to on all platforms.
+///
+/// [ProcessSignal.sigterm] is listened to on all platforms except Windows.
+Future<void> waitForTerminate() {
+  final completer = Completer<bool>.sync();
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply gcp/lib/src/terminate.dart:25]: there's no need for it to be async – so might as well be synchronous.

File: gcp/lib/src/terminate.dart [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      sigIntSub = null;
+      await subCopy.cancel();
+      sigIntSub = null;
+      if (sigTermSub != null) {
+        await sigTermSub!.cancel();
+        sigTermSub = null;
+      }
+      completer.complete(true);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply gcp/lib/src/terminate.dart:45]: 🙏

File: gcp/test/gcp_test.dart [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+Future<TestProcess> _run(
+  String dartScript, {
+  Map<String, String>? environment,
+}) =>
+    TestProcess.start(
+      Platform.resolvedExecutable,
+      [dartScript],
+      environment: environment,
+      includeParentEnvironment: false,
+    );
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply gcp/test/gcp_test.dart:139]: 👍

**Pull request merged**